specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/artifact.json

```diff
@@ -0,0 +1,1799 @@
+            "Int",
+            "Decimal",
+            "Guid",
+            "Boolean",
+            "Date"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "StoredProcedureParameterType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "PaginatedArtifactList": {
+      "description": "A paginated list of Artifacts.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "An array of objects of type Artifact.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Artifact"
+          }
+        },
+        "continuationToken": {
+          "description": "The token used in retrieving the next page.  If null, there are no additional pages.",
+          "type": "string"
+        },
+        "nextLink": {
+          "description": "The link to the next page constructed using the continuationToken.  If null, there are no additional pages.",
+          "type": "string"
+        }
+      }
+    },
+    "ArtifactContentInformation": {
+      "description": "Details of an Artifact Content Information.",
+      "type": "object",
+      "properties": {
+        "contentUri": {
+          "description": "The URI of the content.",
+          "type": "string"
+        },
+        "origin": {
+          "description": "The origin of the Artifact creation request. Available origins are 'ExperimentRun', 'LocalUpload', 'WebUpload', 'Dataset', 'ComputeRecord' and 'Unknown'.",
+          "type": "string"
+        },
+        "container": {
+          "description": "The name of container. Artifacts can be grouped by container.",
+          "type": "string"
+        },
+        "path": {
+          "description": "The path to the Artifact in a container.",
+          "type": "string"
+        }
+      }
+    },
+    "PaginatedArtifactContentInformationList": {
+      "description": "A paginated list of ArtifactContentInformations.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "An array of objects of type ArtifactContentInformation.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ArtifactContentInformation"
+          }
+        },
+        "continuationToken": {
+          "description": "The token used in retrieving the next page.  If null, there are no additional pages.",
+          "type": "string"
+        },
+        "nextLink": {
+          "description": "The link to the next page constructed using the continuationToken.  If null, there are no additional pages.",
+          "type": "string"
+        }
+      }
+    },
+    "ArtifactIdList": {
+      "description": "Contains list of Artifact Ids.",
+      "type": "object",
+      "properties": {
+        "artifactIds": {
+          "description": "List of Artifacts Ids.",
+          "uniqueItems": false,
```

Comment:
Why do you allow duplicate ArtifactIds?

Reply:
This is a tricky one. As mentioned previously, we take dependency on swashbuckle tool to auto generate swaggers for our services. Neither Swashbuckle allow us to set the uniqueItems attribute for the properties nor JsonPropertyattribute offered by famous Newtonsoft.Json c# package enable us to set "uniqueItems". As the scale of the Azure Machine Learning services is very large, It is difficult to maintain any manual modification on the swaggers. I am happy to take in any suggestions.